spec/02-integration/05-proxy/01-router_spec.lua

```diff
@@ -149,6 +149,58 @@ describe("Router", function()
     end)
   end)
 
+  describe("URI arguments (querystring)", function()
+
+    setup(function()
+      insert_apis {
+        {
+          name = "api-1",
+          upstream_url = "http://httpbin.org",
+          hosts = { "example.com" },
+        },
+      }
+
+      assert(helpers.start_kong())
+    end)
+
+    teardown(function()
+      helpers.stop_kong()
+    end)
+
+    it("preserves URI arguments", function()
+      local res = assert(client:send {
+        method = "GET",
+        path   = "/get",
+        query  = {
+          foo   = "bar",
+          hello = "world",
+        },
+        headers = {
+          ["Host"] = "example.com",
+        },
+      })
+
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      assert.equal("bar", json.args.foo)
+      assert.equal("world", json.args.hello)
+    end)
+
+    it("does not proxy '$is_args' if URI does not contain arguments", function()
```

Comment:
So, i did find one case where someone noted that the presence of `?` was semantically significant (granted, not our project or even our lang): https://github.com/requests/requests/issues/2912

In addition, RFC 3986 1.2.3 points out:

``` 
The generic syntax uses the slash ("/"), question mark ("?"), and number sign ("#")
characters to delimit components that are significant to the generic parser's hierarchical
interpretation of an identifier. 
```

And from section 6.2.3:

```
Normalization should not remove delimiters when their associated
component is empty unless licensed to do so by the scheme
specification... the presence or absence of delimiters within a userinfo
subcomponent is usually significant to its interpretation.
```

> If we ever receive such a requirement from a use-case, beer is on me!

http://beerme.pointlessmicroservic.es